google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeployAction.java

```diff
@@ -0,0 +1,248 @@
+    consoleLogLn("Copied %s %s to %s", targetFileName,
+        sourceFilePath.getAbsolutePath(), destinationFilePath.getAbsolutePath());
+    return destinationFilePath;
+  }
+
+  private class DeployToAppEngineProcessListener extends ProcessAdapter {
+    private StringBuilder deployOutput = new StringBuilder();
+    private String version;
+
+    public DeployToAppEngineProcessListener(String version) {
+      this.version = version;
+    }
+
+    @Override
+    public void onTextAvailable(ProcessEvent event, Key outputType) {
+      if(outputType.equals(ProcessOutputTypes.STDOUT)) {
+        deployOutput.append(event.getText());
+      }
+    }
+
+    @Override
+    public void processTerminated(final ProcessEvent event) {
+      if (event.getExitCode() == 0) {
+        callback.succeeded(new DeploymentRuntime() {
+          @Override
+          public boolean isUndeploySupported() {
+            return true;
+          }
+
+          @Override
+          public void undeploy(@NotNull final UndeploymentTaskCallback callback) {
+            SwingUtilities.invokeLater(new Runnable() {
+              @Override
+              public void run() {
+                int doStop = Messages
+                    .showYesNoDialog(
+                        GctBundle.message("appengine.stop.modules.version.confirmation.message"),
+                        GctBundle.message("appengine.stop.modules.version.confirmation.title"),
+                        "OK", "Cancel", General.Warning);
+
+                if (doStop == Messages.YES) {
+                  stop(callback);
+                } else {
+                  callback.errorOccurred("Undeploy Canceled");
+                }
+              }
+            });
+          }
+
+          private void stop(@NotNull UndeploymentTaskCallback callback) {
+            Set<String> modulesToSop;
+            try {
+              modulesToSop = parseDeployOutputToModuleList(deployOutput.toString());
+            } catch (JsonParseException e) {
+              logger.warn("Could not retrieve module(s) of deployed application", e);
+              return;
+            }
+
+            final AppEngineAction appEngineStopAction = appEngineHelper.createStopAction(
+                getLoggingHandler(),
+                modulesToSop,
+                version,
+                callback);
+
+            ProgressManager.getInstance()
+                .run(new Task.Backgroundable(project, "Stop App Engine", true,
+                    null) {
+                  @Override
+                  public void run(@NotNull ProgressIndicator indicator) {
+                    ApplicationManager.getApplication().invokeLater(appEngineStopAction);
+                  }
+                });
+          }
+        });
+
+        SwingUtilities.invokeLater(new Runnable() {
+          @Override
+          public void run() {
+            Messages.showMessageDialog(
+                GctBundle.message("appengine.deployment.success.dialog.message"),
```

Comment:
Yes, the user has to dismiss it. The reason I created this one is to keep it consistent with the stop flow which now has a "successful stop" dialog. This dialog is not completely necessary since the user can see a message of succesfull deployment in the console..